clld/db/schema_migrations/update_unique_null.py

```diff
@@ -0,0 +1,161 @@
+# update_unique_null.py - adapt composite unique constraints and nullables
+
+from __future__ import unicode_literals
+
+# revision identifiers, used by Alembic.
+revision = ''
+down_revision = ''
+
+# INSTRUCTIONS:
+# - cd into the clld app directory
+# - $ alembic revision -m "update assoc tables"
+# - edit migrations/versions/<...>_update_assoc_tables.py
+# - replace the content after 'down_revision = <...>' with the content below
+# - $ alembic upgrade head
+# - check the output of your "update assoc tables" migration (will end with a RuntimeError)
+# - if the changes are as intended, change 'dry=False' to 'dry=True' below
+# - run alembic upgrade head once again to apply them
+
+from alembic import op
+import sqlalchemy as sa
+
+UNIQUE_NULL = [
+    ('contributioncontributor',
+        ['contribution_pk', 'contributor_pk'], []),
+    ('contributionreference',
+        ['contribution_pk', 'source_pk', 'description'],
+        ['description']),
+    ('domainelement',
+        ['parameter_pk', 'name'],
+        ['name']),
+    ('domainelement',
+        ['parameter_pk', 'number'],
+        ['number']),
+    ('editor',
+        ['dataset_pk', 'contributor_pk'], []),
+    ('languageidentifier',
+        ['language_pk', 'identifier_pk'], []),
+    ('languagesource',
+        ['language_pk', 'source_pk'], []),
+    ('sentencereference',
+        ['sentence_pk', 'source_pk', 'description'],
+        ['description']),
+    ('unit',
+        ['language_pk', 'id'],  # FIXME: or rather name?
```

Comment:
`UNIQUE(unit.language_pk, unit.id)` is safe as we already have `UNIQUE(unit.id)`.
TBD: Normally, adding `UNIQUE(unit.language_pk, unit.name)` instead would make sense here (`name` unique within its language if in case it is set at all).

Reply:
Uniqueness of `unit.name` cannot be enforced. E.g. dictionaria stores dictionary headwords as `unit.name`, and only distinguishes homonyms by an additional column `number`.

Reply:
Thanks